types/ioredis/index.d.ts

```diff
@@ -948,14 +949,33 @@ declare namespace IORedis {
         setnx(key: KeyType, value: ValueType, callback: (err: Error, res: number) => void): void;
         setnx(key: KeyType, value: ValueType): Promise<BooleanResponse>;
 
+        setex(key: KeyType, seconds: number, value: ValueType, callback: (err: Error, res: Ok) => void): void;
+        setex(key: KeyType, seconds: number, value: ValueType): Promise<Ok>;
```

Comment:
These are technically wrong, you can provide a callback AND get a promise at the same time, see my PR.

Reply:
Also it would probably make more sense to move these methods into a common interface for Redis and Cluster rather than duplicating them.

Reply:
Ah I see, I could wait for #40887 to be merged, then integrate those changes into this branch?

Also, just to be clear on the common interface, you mean creating a separate interface, and having Redis and Cluster extend them, right?

Reply:
Yes.